spec/Generics.tex

```diff
@@ -246,26 +278,56 @@ \subsection{Formal Arguments of Generic Type}
 \end{chapel}
 \end{example}
 
+\subsection{Formal Arguments of Partially Generic Type}
+\label{Formal_Arguments_of_Partially_Generic_Type}
+\index{formal arguments!partially concrete}
+\index{formal arguments!partially generic}
 \index{where@\chpl{where}!implicit}
-The generic type may be specified with some queries and some exact
-values.  Using exact values in this manner makes the argument
+
+The generic type for a formal argument may be specified with some queries
+or generic types and some concrete types or values.
+Using concrete types or values in this manner makes the argument
 \emph{partially concrete} for the purpose of function resolution.
-\begin{example}
-Given the class definition
+
+\begin{chapelexample}{nested-type-queries.chpl}
+
+Given the class definitions
 \begin{chapel}
 class C {
-  type t;
-  type tt;
+  type elementType;
+  type indexType;
+  type containerType;
+}
+class Container {
+  type containedType;
 }
 \end{chapel}
-then the function definition
+Here, the function \chpl{f} can only apply when the
+\chpl{c.elementType==real}:
```

Comment:
I suggest moving the above statement **after** the definition of `f`, removing "Here," and changing the earlier "Given the class definitions" with "Given the code:" This brings more code together, which I prefer.